oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie.py

```diff
@@ -0,0 +1,337 @@
+                bowtie_results = _read_bowtie_output(batch_id, subbatch_id)
+                num_probes_wo_match += _filter_probes_bowtie(
+                    probes_info, bowtie_results
+                )
+
+        def _read_bowtie_output(batch_id, subbatch_id):
+            """Load the output of the bowtie alignment search into a DataFrame and process the results.
+
+            :return: DataFrame with processed bowtie alignment search results.
+            :rtype: pandas.DataFrame
+            """
+            file_bowtie_batch = os.path.join(
+                self.dir_bowtie, "bowtie_batch{}_{}.txt".format(batch_id, subbatch_id)
+            )
+            bowtie_results = pd.read_csv(
+                file_bowtie_batch,
+                header=None,
+                sep="\t",
+                low_memory=False,
+                names=[
+                    "query",
+                    "strand",
+                    "reference",
+                    "ref_start",
+                    "query_sequence",
+                    "read_quality",
+                    "num_instances",
+                    "mismatch_positions",
+                ],
+                engine="c",
+                dtype={
+                    "query": str,
+                    "strand": str,
+                    "reference": str,
+                    "ref_start": int,
+                    "query_sequence": str,
+                    "read_quality": str,
+                    "num_instances": int,
+                    "mismatch_positions": str,
+                },
+            )
+
+            bowtie_results["query_gene_id"] = (
+                bowtie_results["query"].str.split("_").str[0]
+            )
+            bowtie_results["reference_gene_id"] = (
+                bowtie_results["reference"].str.split("::").str[0]
+            )
+
+            return bowtie_results
+
+        def _filter_probes_bowtie(probes_info, bowtie_results):
+            """Use the results of the Bowtie alignment search to remove probes with high similarity (i.e. low number of mismatches) based on user defined thresholds.
+            :param probes_info: Dataframe with probe information, filtered based on sequence properties.
+            :type probes_info: pandas.DataFrame
+            :param blast_results: DataFrame with processed bowtie alignment search results.
+            :type blast_results: pandas.DataFrame
+            """
+
+            bowtie_results_matches = bowtie_results[
+                bowtie_results["query_gene_id"] != bowtie_results["reference_gene_id"]
+            ]
+
+            if self.ligation_region > 0:
+
+                # Get probes where mismatch positions are given
+                bowtie_results_matches_mismatch_positions = bowtie_results_matches[
+                    bowtie_results_matches["mismatch_positions"].notna()
+                ]
+                # Extract positions (Positions of mismatches given in bowtie output)
```

Comment:
I'm not sure if this is correct because we don't consider if the match is on the + or - strand. If it's on the minus strand the missmatches should be counted on the reverse complement